src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubectlBuildWrapper.java

```diff
@@ -0,0 +1,129 @@
+
+    @Override
+    public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener, EnvVars initialEnvironment) throws IOException, InterruptedException {
+
+        FilePath configFile = workspace.createTempFile(".kube", "config");
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " set-cluster k8s --server=" + serverUrl + " --insecure-skip-tls-verify=true")
+                .join();
+
+        final UsernamePasswordCredentials c = getCredentials();
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " set-credentials cluster-admin --username=" + c.getUsername() + " --password=" + Secret.toString(c.getPassword()))
+                .masks(false, false, false, false, false, false, true)
+                .join();
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " set-context k8s --cluster=k8s --user=cluster-admin")
+                .join();
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " use-context k8s")
+                .join();
+
+        context.setDisposer(new CleanupDisposer(configFile));
+
+        context.env("KUBECONFIG", configFile.getRemote());
+    }
+
+    private UsernamePasswordCredentials getCredentials() {
+        return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(StandardUsernamePasswordCredentials.class,
+                        Jenkins.getInstance(), ACL.SYSTEM, Collections.<DomainRequirement>emptyList()),
+                CredentialsMatchers.withId(credentialsId)
+        );
+
+    }
+
+    @Extension
+    public static class DescriptorImpl extends BuildWrapperDescriptor {
+
+        @Override
+        public boolean isApplicable(AbstractProject<?, ?> item) {
+            return true;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Setup Kubernetes CLI (kubectl)";
+        }
+
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item item, @QueryParameter String serverUrl) {
+            return new StandardListBoxModel()
+                    .withEmptySelection()
+                    .withMatching(
+                            CredentialsMatchers.instanceOf(StandardUsernamePasswordCredentials.class),
+                            CredentialsProvider.lookupCredentials(
+                                    StandardCredentials.class,
+                                    item,
+                                    null,
+                                    URIRequirementBuilder.fromUri(serverUrl).build()
+                            )
+                    );
+
+        }
+
+    }
+
+    private static class CleanupDisposer extends Disposer {
+
+        private FilePath configFile;
```

Comment:
While there is a `FilePathPickle`, I would not rely on it for serialization. Just save the `String` relative path in the workspace.